### PR TITLE
Improve missforest interface

### LINKED ISSUE #279: Nicer interface for miss_forest_impute
**Is your feature request related to a problem? Please describe.**

`var_names: dict[str, list[str]] | None = None` is the input parameter for miss_forest_impute. However, for all other imputation methods it is `var_names: list[str] | None = None`. 

**Describe the solution you would like**

Let's add support for only a list of strings also to miss_forest_impute. We will simply make the assumption that we are only imputing non-mixed data. Just wrap this with a try/except statement and if it crashes denote that a different parameter format is required for mixed data.

### PR DESCRIPTION
**Pull request by Zethson**
Fixes #279 and #283

@Imipenem The dict interface of MissForest is currently not tested, right?

### CODE REVIEW

File: ehrapy/api/preprocessing/_data_imputation.py [hunk 1]
```diff
@@ -5,11 +5,12 @@
 from anndata import AnnData
 from rich import print
 from rich.progress import Progress, SpinnerColumn
-from sklearn.ensemble import ExtraTreesRegressor, RandomForestClassifier
 from sklearn.experimental import enable_iterative_imputer  # noqa: F401
-from sklearn.impute import IterativeImputer, KNNImputer, SimpleImputer
```
Comment (Imipenem) [inline ehrapy/api/preprocessing/_data_imputation.py:10]: are those two removed?

Comment (Zethson) [reply ehrapy/api/preprocessing/_data_imputation.py:10]: Locally imported after the Intel sklearn patch.

**Comment by Imipenem** [comment]
General question: Does this work on other CPU architectures as well? Only asking because it's called `ìntel...`?

**Comment by Zethson** [comment]
> General question: Does this work on other CPU architectures as well? Only asking because it's called `ìntel...`?

I hope so. I'll check again.

**Comment by Zethson** [comment]
image

Should also support AMD.

**Comment by Zethson** [comment]
Apparently a shitshow to install on M1 macs. I will add this as an optional dependency and only use the import when it's installed.

**Pull request merged**